wireapp/wire-android
# feat: proxy support [AR-2280]

<table>
<td>
  <a href="https://wearezeta.atlassian.net/browse/AR-2280" title="AR-2280" target="_blank"><img alt="Story" src="https://wearezeta.atlassian.net/rest/api/2/universal_avatar/view/type/issuetype/avatar/10815?size=medium" />AR-2280</a>  Proxy support
  </td></table>
  <br />
 














----
#### PR Submission Checklist for internal contributors

- The **PR Title**

- The **PR Description**
----

# What's new in this PR?

### Issues

to add another layer of security we needed to add the support to add a socks proxy while logging in


### Solutions

detect the proxy from the configs and update the login screen to make the user able to insert the proxy credentials


Needs releases with:
https://github.com/wireapp/kalium/pull/1037

### Testing


#### How to Test

update the server config with proxy host and port (should work through deeplink if proxy was specified there), and continue the login process with filling the proxy credentials section.

### Notes (Optional)

Testing can be a bit tricky for this one, so if any help needed please approach me and I can help with that.

### Attachments (Optional)

_Attachments like images, videos, etc. (drag and drop in the text box)_ 



----
#### PR Post Submission Checklist for internal contributors (Optional)

----
#### PR Post Merge Checklist for internal contributors

----
##### References
1. https://sparkbox.com/foundry/semantic_commit_messages
1. https://github.com/wireapp/.github#usage
1. E.g. `feat(conversation-list): Sort conversations by most emojis in the title #SQPIT-764`.
